pkg/kubelet/kubelet.go

```diff
@@ -2014,14 +2014,27 @@ func (kl *Kubelet) SyncTerminatingPod(_ context.Context, pod *v1.Pod, podStatus
 
 	kl.probeManager.StopLivenessAndStartup(pod)
 
-	p := kubecontainer.ConvertPodStatusToRunningPod(kl.getRuntime().Type(), podStatus)
-	if err := kl.killPod(ctx, pod, p, gracePeriod); err != nil {
+	var podStopped bool
+	var err error
+	if !utilfeature.DefaultFeatureGate.Enabled(features.RestartContainerDuringTermination) {
+		p := kubecontainer.ConvertPodStatusToRunningPod(kl.getRuntime().Type(), podStatus)
+		err = kl.killPod(ctx, pod, p, gracePeriod)
```

Comment:
If the Feature Gate is off, podStopped will always be false, which I think will always cause an error saying the pod is not yet stopped at line 2035 and change the current behavior.

```suggestion
		err = kl.killPod(ctx, pod, p, gracePeriod)
		if err == nil {
			podStopped = true
		}
```